sirepo/package_data/static/js/controls.js

```diff
@@ -103,13 +109,63 @@ SIREPO.app.controller('ControlsController', function(appState, controlsService,
         };
     }
 
+    function updateFromMonitorValues(monitorValues) {
+        monitorValues.forEach((value) => {
+            // TODO(e-carlin): How can I eliminate the timeout. Needed because
+            // on first load self.watches needs to be populated before there is
+            // anyone listening to the broadcast
+            $timeout(function () {
+                // TODO(e-carlin): need a better connection between element name,
+                // monitor values and the bpmMonitorPlots
+                $scope.$broadcast(
+                    'sr-pointData-' + elementForName(value.name).name,
+                    [value.x, value.y]
+                );
+            });
+        });
+    }
+
+    self.simHandleStatus = function(data) {
+        if (data.monitorValues) {
+            frameCache.setFrameCount(1);
+            updateFromMonitorValues(data.monitorValues);
+        }
+    };
+
     appState.whenModelsLoaded($scope, function() {
+        self.commandEditors = [];
         self.editorColumns = [];
+        self.watches = [];
+        // TODO(e-carlin): twiss_command the fields aren't all populated correctly
```

Comment:
@moellep there are some problems with the model being nested under externalLattice. How should I resolve? Should I make the concept of externalLattice known to sirepo-components.js and sirepo.js so it looks for the models under it if it isn't found at the top level?

Reply:
We don't want to push the externalLattice info outside of the controls app. I changed the code so the beam and twiss models get edited as a copy in appState.models and then the externalLattice gets updated whenever the models change.